src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Implementation.cs

```diff
@@ -249,6 +249,7 @@ private async Task ForceAuthenticationAsync<TIOAdapter>(TIOAdapter adapter, bool
                     if (message.Size > 0)
                     {
                         await adapter.WriteAsync(message.Payload!, 0, message.Size).ConfigureAwait(false);
+                        await adapter.FlushAsync().ConfigureAwait(false);
```

Comment:
There's another write below in a failure path. That one doesn't need to be flushed?

Reply:
I'm not sure @stephentoub. It will not create deadlock described in #37380 but it is probably good idea.